x-pack/packages/kbn-langchain/server/language_models/chat_openai.ts

```diff
@@ -207,20 +216,38 @@ export class ActionsClientChatOpenAI extends ChatOpenAI {
         ...('tool_call_id' in message ? { tool_call_id: message?.tool_call_id } : {}),
       })),
     };
+    const subAction =
+      llmType === 'inference'
+        ? completionRequest.stream
+          ? 'completion_stream'
+          : 'completion'
+        : // langchain expects stream to be of type AsyncIterator<OpenAI.ChatCompletionChunk>
+        // for non-stream, use `run` instead of `invokeAI` in order to get the entire OpenAI.ChatCompletion response,
+        // which may contain non-content messages like functions
+        completionRequest.stream
+        ? 'invokeAsyncIterator'
+        : 'run';
     // create a new connector request body with the assistant message:
+    const subActionParams =
+      llmType === 'inference'
+        ? {
+            input: body.messages.reduce((acc, i) => {
```

Comment:
combining all the messages since the inference endpoint currently only takes string arg. Sending a long message does not work though, I only got this to work by sending "hello world"